src/main/java/seedu/address/ui/AttendanceRow.java

```diff
@@ -0,0 +1,36 @@
+package seedu.address.ui;
+
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+
+import seedu.address.model.student.Student;
+
+/**
+ * Represents a row in the attendance table.
+ */
+public class AttendanceRow {
+    private final String studentName;
+    private final Map<LocalDate, String> attendanceByDate;
+
+    /**
+     * Creates an AttendanceRow object for the specified student.
+     * @param student
+     */
+    public AttendanceRow(Student student) {
+        this.studentName = student.getName().fullName;
+        this.attendanceByDate = new HashMap<>();
+    }
+
+    public String getStudentName() {
+        return studentName;
+    }
+
+    public void addAttendance(LocalDate date, String attendance) {
+        attendanceByDate.put(date, attendance);
+    }
+
+    public String getAttendanceForDate(LocalDate date) {
+        return attendanceByDate.getOrDefault(date, "Absent"); // Default to "Absent" if no attendance record
```

Comment:
Should we just specify that there is no attendance record as the default value, i.e. "No record found"? That way, the tutor would know that he hasn't marked the attendance for that student yet, when checking through his records.

Reply:
Hmm, good suggestion.

I think since attendance only has 2 choices (present / absent),  it is safe to say that anyone who isn't specifically marked present will just be counted as absent.

Will also wait on the implementation of Mark_Present_All and Unmark_Present_All to see how we want to change the default value

Reply:
I feel like we might need a way to delete an attendance record in the case that the tutor adds attendance for the wrong date (where they don't have tutorial), since currently we can only change an existing attendance record to be from present to absent.

Reply:
Sounds good. Will implement that in 1.5